frontend/src/metabase/embedding-sdk/components/SimpleDataPicker/SimpleDataPicker.tsx

```diff
@@ -1,80 +1,90 @@
+  isInitiallyOpen,
+  setSourceTableFn,
+  triggerElement,
 }: SimpleDataPickerProps) {
-  const shouldShowSearchBar = options.length > 10;
-  const [searchText, setSearchText] = useState("");
-  function filterSearch(option: Options): boolean {
-    if (searchText) {
-      return normalizeString(option.display_name).includes(
-        normalizeString(searchText),
-      );
+  const [isDataPickerOpened, { toggle, close }] =
+    useDisclosure(isInitiallyOpen);
+  const { data, isLoading, error } = useSearchQuery({
+    models: ["dataset", "table"],
+  });
+
+  const options = useMemo(() => {
+    if (!data) {
+      return [];
     }
 
-    return true;
-  }
+    return sortEntities(
+      data.data.map(entity => {
+        return {
+          ...entity,
+          id: isModel(entity)
+            ? getQuestionVirtualTableId(entity.id)
+            : entity.id,
+        };
+      }),
+      { sort_column: "name", sort_direction: SortDirection.Asc },
+    );
+  }, [data]);
 
   return (
-    <Paper w="300px" p="sm">
-      {shouldShowSearchBar ? (
-        <TextInput
-          data-autofocus
-          type="search"
-          icon={<Icon name="search" size={16} />}
-          mb="sm"
-          placeholder={t`Search…`}
-          onChange={e => setSearchText(e.target.value ?? "")}
-        />
-      ) : (
-        /**
-         * Behave like Mantine 7's `FocusTrap.InitialFocus`.
-         *
-         * This component disable the automatic focus on the first focusable element
-         * when there is no search bar.
-         */
-        <span aria-hidden data-autofocus tabIndex={-1} />
-      )}
-      {/* @ts-expect-error - I think the typing for ScrollArea.Autosize is wrong. This might be fixed in Mantine 7 */}
-      <ScrollArea.Autosize mah={TEN_OPTIONS_HEIGHT} type="auto">
-        {options.filter(filterSearch).map(option => {
-          return (
-            <NavLink
-              key={option.id}
-              active={selectedEntity === option.id}
-              icon={<Icon c="var(--mb-color-icon-primary)" name="table" />}
-              label={option.display_name}
-              onClick={() => {
-                onClick(option);
-              }}
-              variant="default"
-            />
-          );
-        })}
-      </ScrollArea.Autosize>
-    </Paper>
+    <Popover
+      opened={isDataPickerOpened}
+      position="bottom-start"
+      onClose={close}
+    >
+      <Popover.Target>
+        <Box onClick={toggle}>{triggerElement}</Box>
+      </Popover.Target>
+      <Popover.Dropdown>
+        <DelayedLoadingAndErrorWrapper loading={isLoading} error={error}>
+          <SimpleDataPickerView
+            selectedEntity={selectedEntity}
+            onClick={setSourceTableFn}
+            options={options}
+          />
+        </DelayedLoadingAndErrorWrapper>
+      </Popover.Dropdown>
+    </Popover>
   );
 }
 
-function normalizeString(input: string) {
-  return input.trim().toLowerCase();
+function sortEntities(
```

Comment:
use a similar sorting function as https://github.com/metabase/metabase/blob/75e9be30f65037cb81089ed94ff3df9fdc93b70b/frontend/src/metabase/browse/models/utils.ts#L48-L75

I wanted to reuse it initially, but the type isn't compatible and there's no guarantee that that function will be modified since model browse and entity picker are different domains, so I implement a new sort function.

Reply:
I guess we already have an e2e that covered the sorted results so we don't need to unit-test `sortEntities`, right?

Reply:
well, no, we don't have tests for this. I'll add unit tests since it's too heavy to do E2E 🙏

Reply:
@WiNloSt perfect then!